plutus-ledger-constraints/src/Ledger/Constraints/TxConstraints.hs

```diff
@@ -555,7 +555,11 @@ mustSpendPubKeyOutput = singleton . MustSpendPubKeyOutput
 -- If used in 'Ledger.Constraints.OnChain', this constraint verifies that the
 -- transaction spends this @utxo@.
 mustSpendScriptOutput :: forall i o. TxOutRef -> Redeemer -> TxConstraints i o
-mustSpendScriptOutput txOutref = singleton . MustSpendScriptOutput txOutref
+mustSpendScriptOutput txOutref red = singleton $ MustSpendScriptOutput txOutref red Nothing
+
+mustSpendScriptOutputWithReference :: TxOutRef -> Redeemer -> TxOutRef -> TxConstraints i o
```

Comment:
Add a comment explaining what it does, like the ones above.

Reply:
🤦